### PR TITLE
73 Add schema option for Dataset comparison

### LINKED ISSUE #73: Add option to specify which schema to use
## Feature
Add option to specify which schema to use, thus comparing only fields of interest.

### PR DESCRIPTION
**Pull request by Zejnilovic**
This commit will allow the user to specify path to a schema on HDFS that they want to use. This will then allow them to cherry-pick column for dataset comparison.

Closes #73

### CODE REVIEW

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparison.scala [hunk 1]
```diff
@@ -145,6 +149,23 @@ class DatasetComparison(cliOptions: CliOptions,
     }
   }
 
+  /**
+   * Performs a check if the schemas of two data frames are supersets of schema provided..
+   *
+   * @param testedDF Comparison pair of two DataFrames whose schema will be tested
+   * @param schema Schema that needs to be a subset of schemas provided by data sets
+   */
+  def checkSchemas(testedDF: ComparisonPair[DataFrame], schema: StructType): Unit = {
```
Comment (benedeki) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparison.scala:157]: I would name the function little differently. Same name function overloaded is when they provide same functionality, here it differs somewhat.

**Comment by Zejnilovic** [comment]
image

@yruslan @benedeki @HuvarVer

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala [hunk 2]
```diff
@@ -47,7 +48,13 @@ object DatasetComparisonJob {
   def execute(cliOptions: CliOptions, configPath: Option[String] = None)
              (implicit sparkSession: SparkSession): Unit = {
     val config = new TypesafeConfig(configPath)
-    val dsComparison = new DatasetComparison(cliOptions, config)
+    val optionalSchema = cliOptions.schemaPath match {
+      case Some(schema) =>
```
Comment (benedeki) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:52]: Would rename the `schema` to `schemaPath` or such. It's not the schema yet.

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala [hunk 3]
```diff
@@ -47,7 +48,13 @@ object DatasetComparisonJob {
   def execute(cliOptions: CliOptions, configPath: Option[String] = None)
```
Comment (benedeki) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:48]: Would make this thing more functional. Make the `execute` function return `result` and do the reporting "at the end of the world".

**Review by benedeki — COMMENTED** [review]
Later for this one too. Sorry 😊

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala (hunk 2 above)
Comment (Zejnilovic) [reply datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:52]: Fair enough

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparison.scala (hunk 1 above)
Comment (Zejnilovic) [reply datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparison.scala:157]: I do view them as same. Checking that data frames are withing the schema. In the first one, the schema is "implicit" in a sense. that is taken from data frames.

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala (hunk 3 above)
Comment (Zejnilovic) [reply datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:48]: Method execute is the world. DatasetComparison#compare returns a result and the job is defined to handle. the result like this. I am basically allowing users for 3 entry points into the comparison. SparkSubmit from the command line, use method execute to skip over parsing and have only some pre-processing or DatasetComparison as a full-fledged library.

**Pull request merged**